test/server/cardImplementations/01_SOR/AvengerHuntingStarDestroyer.spec.js

```diff
@@ -0,0 +1,91 @@
+                this.cartelSpacer = this.player2.findCardByName('cartel-spacer');
+                this.p2Avenger = this.player2.findCardByName('avenger#hunting-star-destroyer');
+
+                this.p1Base = this.player1.base;
+                this.p2Base = this.player2.base;
+
+                this.noMoreActions();
+            });
+
+            it('forces opponent to defeat friendly non-leader unit when Avenger is played', function () {
+                // Play Avenger
+                this.player1.clickCard(this.p1Avenger);
+
+                // Player 2 must choose its own unit
+                expect(this.player2).toBeAbleToSelectAllOf([this.wampa, this.cartelSpacer, this.p2Avenger]);
+                expect(this.player2).toBeAbleToSelectNoneOf([this.p1Avenger, this.interceptor, this.pykeSentinel]);
+                this.player2.clickCard(this.cartelSpacer);
+
+                // Chosen unit defeated
+                expect(this.cartelSpacer.location).toBe('discard');
+            });
+
+            it('forces opponent to defeat friendly non-leader unit when Avenger attacks', function () {
+                this.player2.setActivePlayer();
+
+                // Attack with Avenger, choose base as target
+                this.player2.clickCard(this.p2Avenger);
+                this.player2.clickCard(this.p1Base);
+
+                // Player 1 must choose its own unit
+                expect(this.player1).toBeAbleToSelectAllOf([this.interceptor, this.pykeSentinel]);
+                expect(this.player1).toBeAbleToSelectNoneOf([this.wampa, this.cartelSpacer, this.p2Avenger]);
+                this.player1.clickCard(this.pykeSentinel);
+                expect(this.pykeSentinel.location).toBe('discard');
+                expect(this.p1Base.damage).toBe(8);
+            });
+
+            it('allows the defender to be defeated and end the attack', function () {
+                this.player2.setActivePlayer();
+
+                // Attack with Avenger, choose interceptor as target
+                this.player2.clickCard(this.p2Avenger);
+                this.player2.clickCard(this.interceptor);
+
+                // Interceptor not yet destroyed
+                expect(this.interceptor.location).toBe('space arena');
+
+                // Player 1 must choose its own unit
+                expect(this.player1).toBeAbleToSelectAllOf([this.interceptor, this.pykeSentinel]);
+                expect(this.player1).toBeAbleToSelectNoneOf([this.wampa, this.cartelSpacer, this.p2Avenger]);
+
+                // Choose the defender and check it was destroyed
+                this.player1.clickCard(this.interceptor);
+                expect(this.interceptor.location).toBe('discard');
+
+                // Ensure no damage happened
+                expect(this.p2Avenger.damage).toBe(0);
+                expect(this.p1Base.damage).toBe(0);
+            });
+
+            // it('Deployed leader units may not be chosen for Avenger ability', function () {
```

Comment:
Can you add a comment here that just says `TODO LEADERS` so we know to come back here and fix it when we do that work